src/ServiceBus.AttachmentPlugin/AzureStorageAttachment.cs

```diff
@@ -96,7 +117,7 @@ static void SetValidUntil(ICloudBlob blob, TimeSpan timeToBeReceived)
             }
             else
             {
-                var container = client.Value.GetContainerReference(configuration.ContainerName);
+                var container = client.GetContainerReference(configuration.ContainerName);
```

Comment:
This will require `await InitializeClient().ConfigureAwait(false);` to ensure the client is not `null`. Similar to what `BeforeMessageSend` is doing to cover scenarios where a plugin is used to receive only w/o SAS uri. I'll add it after merge.